File: AddGenes/buildHomologsDB.py
import os,re

orthomclDB = open("OrthoMCL.fasta", "r").readlines()
ogs = open("oglistTest", "r").readlines()
os.system("mkdir -p Data/OGs")

tag = ""
seq = ""
line = ""
orthomclHash = {}

for line in orthomclDB:
	line = line.replace("\n", "")
	
	if ">" in line:
		print line
		if not seq == "" : orthomclHash[tag] = seq
		tag = line
		seq = ""
	
	if re.match("(^[A-z])|(^\*)", line):
		seq = seq + line

orthomclHash[tag] = seq

for og in ogs:
	og = og.replace("\n", "")
	ogFile = open("Data/OGs/" + og, "w")
	print og + "\n"
	
	for key in orthomclHash:
		if og in key:
			ogFile.write(key + "\n" + orthomclHash[key] + "\n")
	ogFile.close()

File: PhyloToL_v3/AddGenes/buildHomologsDB.rb
orthomclDB = File.open("OrthoMCL.fasta", "r").readlines()
ogs = File.open("oglistTest", "r").readlines()
`mkdir -p Data/OGs`

tag = ''
seq = ''
line = ''
orthomclHash = Hash.new

orthomclDB.each do |line|
	line = line.gsub("\n", "")
	
	if line =~ />/
		puts line
		if not seq == "" then orthomclHash[tag] = seq end
		tag = line
		seq = ""
	end
	
	if line =~ /(^[A-z])|(^\*)/
		seq = seq + line
	end
end 

orthomclHash[tag] = seq

ogs.each do |og|
	og = og.gsub("\n", "")
	ogFile = File.open("Data/OGs/" + og, "w")
	puts og + "\n"
	
	orthomclHash.each do |key, value|
		if key.include? og
			ogFile.write(key + "\n" + value + "\n")
		end
	end
	ogFile.close
end
